Bottom: bd0093f2f16e5fa8c86710a942e8c65dbb3017c6
Top:    b084f45ddf4841fbed9f6a73b819cd1a3fa354ff
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 01:46:21 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 040124c..545aaac 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1136,7 +1136,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
